### PR TITLE
Switch BCI images

### PR DESCRIPTION
**Pull request by bk201**
- Build environment:
  - Switch to BCI image and use zypper to install packages.
  - Drop `GO111MODULE=off`.
  - Add go pkg and cache volumes.
- harvester-cluster-repo: switch to BCI image and install nginx.

**Related issue**
https://github.com/harvester/harvester/issues/1868

### CODE REVIEW

File: package/harvester-repo/Dockerfile [hunk 1]
```diff
@@ -1,3 +1,15 @@
...
+    ln -sf /dev/stdout /var/log/nginx/access.log && \
+    ln -sf /dev/stderr /var/log/nginx/error.log
+
+COPY charts /srv/www/htdocs/charts
+
+EXPOSE 80
+
+STOPSIGNAL SIGQUIT
+
+CMD ["nginx", "-g", "daemon off;"]
```
Comment (weihanglo) [inline package/harvester-repo/Dockerfile:15]: Reference for reviewers: https://github.com/nginxinc/docker-nginx/blob/b0e153a1b644ca8b2bd378b14913fff316e07cf2/stable/debian/Dockerfile#L110-L114

**Review by weihanglo — COMMENTED** [review]
Looks good and succeeded to install a new cluster from this change.

File: Dockerfile.dapper [hunk 2]
```diff
@@ -16,8 +16,9 @@ ENV HARVESTER_INSTALLER_OFFLINE_BUILD=$HARVESTER_INSTALLER_OFFLINE_BUILD
 ARG DAPPER_HOST_ARCH
 ENV ARCH $DAPPER_HOST_ARCH
 
-RUN apk -U add bash git gcc musl-dev docker vim less file curl wget ca-certificates zstd squashfs-tools xorriso
-RUN apk add yq --repository=http://dl-cdn.alpinelinux.org/alpine/edge/community
+RUN zypper -n rm container-suseconnect && \
+    zypper -n install go1.16 git curl docker gzip tar wget zstd squashfs xorriso awk
+RUN curl -sfL https://github.com/mikefarah/yq/releases/download/v4.14.1/yq_linux_${ARCH} -o /usr/bin/yq && chmod +x /usr/bin/yq
```
[the remark this answers is not in the document]
Comment (bk201) [reply Dockerfile.dapper]: No problem, let's use the latest one.

**Review by weihanglo — APPROVED** [review]
LGTM

**Review by johnliu55tw — APPROVED** [review]
LGTM

**Pull request merged**